### PR TITLE
Add migrating from Logback/SLF4J pages

### PR DESCRIPTION
**Pull request by vy**
This work addresses #2514.

* `migration-from-logback.adoc` is added
* `migration-from-slf4j.adoc` is added
* `concepts.adoc` is added
* `getting-started.adoc` is moved to `manual`
* `Resources` is pushed down further in the navigation, effectively moving `Manual` up (as requested in #1469)
* `getting-started.adoc` is overhauled to accommodate needed links in the new migration guides
   * Installation-vs-Configuration clearly separated
   * Application- and Library-related parts are clearly isolated (Library installation was referring to the application installation, where application installation was stating you can skip this section and proceed to library installation)
   * Consolidated duplicate code-related context both in prose and callouts to only callouts

### CODE REVIEW

File: src/site/antora/modules/ROOT/pages/migrate-from-slf4j.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+====
+
+[#migrating]
+== Migrating
+
+If your application or library logs using SLF4J, you can migrate it to Log4j API as follows:
+
+. Remove `org.slf4j:slf4j-api` dependency
+. Follow the instructions shared in the _"Getting started"_ page for xref:manual/getting-started.adoc#using-api[installing `log4j-api` and using it]
+. Search for `org.slf4j` usages in your project and replace them with Log4j API equivalents
```
[the remark this answers is not in the document]
Comment (vy) [reply src/site/antora/modules/ROOT/pages/migrate-from-slf4j.adoc]: Fixed in 80dc155342cc9fb2f89748543ec5a1709f25f95b.

File: src/site/antora/modules/ROOT/pages/migrate-from-slf4j.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+then you don't need to take any extra steps.
+Unlike applications, libraries should be logging implementation agnostic.
+That is, libraries should log through a logging API, but leave the decision of the logging implementation to the application.
+
+If it is an application that you migrated, and you are not using Log4j Core as your logging implementation,::
+then you can consider doing so.
+Certain xref:manual/api.adoc[Log4j API features] (e.g., xref:manual/garbagefree.adoc[garbage-free logging]) require an end-to-end setup which is mostly possible using Log4j API in combination with Log4j Core.
++
+See xref:manual/installation.adoc[] for installing Log4j Core. +
+If you are using Logback, refer to xref:migrate-from-logback.adoc[].
```
[the remark this answers is not in the document]
Comment (vy) [reply src/site/antora/modules/ROOT/pages/migrate-from-slf4j.adoc:84]: > Can we add a reference to OpenRewrite

What do you mean? There is already one in an `[IMPORTANT]` block.

> for Lombok users

Fixed in 80dc155342cc9fb2f89748543ec5a1709f25f95b.

File: src/site/antora/modules/ROOT/pages/migrate-from-logback.adoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+====
+
+[#migrating]
+== Migrating
+
+You either have an application using Logback at runtime, or have a library using Logback for tests.
+In either case, you can replace Logback with Log4j Core as follows:
+
+. Remove `ch.qos.logback:logback-classic` dependency
```
[the remark this answers is not in the document]
Comment (vy) [reply src/site/antora/modules/ROOT/pages/migrate-from-logback.adoc:44]: Text already includes a mention about SLF4J-to-Log4j bridge. I added the removal of `log4j-to-slf4j` in 80dc155342cc9fb2f89748543ec5a1709f25f95b.

File: src/site/antora/modules/ROOT/pages/migrate-from-slf4j.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/migrate-from-slf4j.adoc:84]: > What do you mean? There is already one in an `[IMPORTANT]` block.

I must have missed it (I am using Github to see the formatted page).

File: src/site/antora/modules/ROOT/pages/migrate-from-logback.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/migrate-from-logback.adoc:44]: I added additional conversion rules in 00139dcdb3ab2d71d5cbb6f17dab6a0e5bd84ecf.

**Review by ppkarwasz — APPROVED** [review]
LGTM for now.

Once #2080 is finished, we will probably also have a configuration conversion tool.

**Pull request merged**